src/FilePositionMap.php

```diff
@@ -0,0 +1,146 @@
+<?php declare(strict_types=1);
+
+namespace Microsoft\PhpParser;
+
+/**
+ * FilePositionMap can be used to get the line number for a large number of nodes (starting from 1).
+ * It works the most efficiently when the requested node is close to the previously requested node.
+ *
+ * Other designs that weren't chosen:
+ * - Precomputing all of the start/end offsets when initializing was slower - Some offsets weren't needed, and walking the tree was slower.
+ * - Caching line numbers for previously requested offsets wasn't really necessary, since offsets are usually close together and weren't requested repeatedly.
+ *
+ * Future optimizations:
```

Comment:
Empty "Future optimizations"?

Reply:
Those became optimizations that I tried (but didn't actually improve performance).

Forgot to remove the section.